starboard/elf_loader/evergreen_info.cc

```diff
@@ -60,8 +60,6 @@ bool GetEvergreenInfo(EvergreenInfo* evergreen_info) {
   // Make sure all memory changes are visible to the current thread.
   SbAtomicMemoryBarrier();
   if (!g_valid_info) {
-    SB_LOG(ERROR) << "g_evergreen_info not yet set";
```

Comment:
I'd rather not remove this log statement. I added it just last week after spending a couple hours debugging an issue that was very difficult to identify without having it. When Cobalt is run in Evergreen mode, this - GetEvergreenInfo() getting called before SetEvergreenInfo() - is a real failure scenario that we want to surface.

I'd like to solve this by instead figuring out a way to ensure GetEvergreenInfo() isn't called when we aren't running Evergreen. I can take a look and see if there's a good way to do that.

Reply:
I suggest changing the log into a `LOG(FATAL)` so that instead of spamming the logs and passing tests, makes builds fail when it triggers.

The logspam is pretty large on `dom_test` in the linux builder. Not reproduced locally, so the CI must be building it differently somehow...

Reply:
I don't think it's safe to always consider this a fatal (i.e., unrecoverable) failure, though. Even for Evergreen, there may be some cases where `GetEvergreenInfo()` ends up getting called before the Evergreen shared library is loaded and evergreen info has been set, but where we don't want to terminate the application. For example, `SbLogRawDumpStack()` could get called before the Evergreen shared lib is loaded.

And for the unit test you mentioned, wouldn't it be an incorrect result if we failed it because of the missing Evergreen info?

The root of the problem here is that `sb_is_evergreen_compatible`, and it's associated C++ macro, is just meant to signify that the platform is **able** to be used as an EG loader platform, not that it is **actively** being used as an EG loader platform in the current build:

```
# Whether this is an Evergreen compatible platform. A compatible platform
# can run the elf_loader and launch the Evergreen build.
sb_is_evergreen_compatible = false
```

It doesn't really make sense to guard things like calls to `GetEvergreenInfo()` on that variable/macro. We really want to branch on some variable/macro that captures "is this loader compatible platform currently being used to load an evergreen binary." @y4vor pointed out offline that we already have a variable like this, but just for Android: `sb_evergreen_compatible_package`. So we could consider generalizing this variable. Based on what @andrewsavage1 said over chat, we already have something like this planned for the main branch and it may not be worth the effort on a stable release branch.

Unless anyone else has an idea, I guess I'm ok with removing the log statement. I still am not thrilled about that but I understand that it's a problem to pollute the unit test logs.

Reply:
If this log is desired to remain, I suggest re-introducing it with a fix that doesn't re-introduce the unit test logspam.